FaunaDB.Client/Client/FaunaClient.cs

```diff
@@ -184,23 +184,7 @@ internal static void RaiseForStatusCode(RequestResult resultRequest)
 
             var response = new QueryErrorResponse(statusCode, wrapper.Errors);
 
-            switch (statusCode)
-            {
-                case 400:
-                    throw new BadRequest(response);
-                case 401:
-                    throw new Unauthorized(response);
-                case 403:
-                    throw new PermissionDenied(response);
-                case 404:
-                    throw new NotFound(response);
-                case 500:
-                    throw new InternalError(response);
-                case 503:
-                    throw new UnavailableError(response);
-                default:
-                    throw new UnknowException(response);
-            }
+            throw ExceptionResolver.Resolve(response, statusCode).FirstOrDefault();
```

Comment:
did we ever handle the case where the response body was not valid JSON? We need to handle that here, throwing the correct exception based on the HTTP status code as a fallback. See the Java driver for an example.